plugins/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/AttachmentProcessor.java

```diff
@@ -119,7 +119,12 @@ public void execute(IngestDocument ingestDocument) {
 
             if (properties.contains(Property.CONTENT_LENGTH)) {
                 String contentLength = metadata.get(Metadata.CONTENT_LENGTH);
-                String length = Strings.hasLength(contentLength) ? contentLength : String.valueOf(parsedContent.length());
+                long length;
+                if (Strings.hasLength(contentLength)) {
+                    length = Long.parseLong(contentLength);
```

Comment:
Maybe `try {length = Long.parseLong(contentLength);} catch (ParseException e) {logger.warn("Invalid content length header [{}] in [{}], using length of parsed document [{}] instead", e, contentLength, someIdentifierIdunnoWhat, parsedContent.length()); length = parseContent.length();}`?

I'm not sure we should blow up entirely if the header is garbage.